src/Symfony/Component/Messenger/Attribute/AsMessage.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Attribute;
+
+/**
+ * Attribute for configuring message routing.
+ *
+ * @author Pierre Rineau pierre.rineau@processus.org>
+ */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
+class AsMessage
+{
+    public function __construct(
+        /**
+         * Name of the transports to which route the message.
+         */
+        public null|string|array $transport = null,
```

Comment:
what's the point of providing no transports?

Reply:
Cf. my message below, the idea is not only to configure transports, but having the `#[AsMessage]` attribute as a marker than can be used for other purpose.

For example, we could imagine a defensive feature where some middleware could use this to forbid completely non message classes to be hydrated and propagated to the bus.

We could also image some custom project using it for targeting message classes and generating documentation.